cli/src/config.rs

```diff
@@ -16,56 +16,118 @@
 
 use exonum::{
     blockchain::{ConsensusConfig, ValidatorKeys},
-    keys::Keys,
+    events::NetworkConfiguration,
+    exonum_merkledb::DbOptions,
+    keys::{read_keys_from_file, Keys},
+    node::{ConnectListConfig, MemoryPoolConfig, NodeApiConfig, NodeConfig as CoreNodeConfig},
 };
+use exonum_supervisor::mode::Mode as SupervisorMode;
 use serde_derive::{Deserialize, Serialize};
 
-use std::{net::SocketAddr, path::PathBuf};
-
-/// Base config.
-#[derive(PartialEq, Clone, Debug, Serialize, Deserialize, Default)]
-pub struct CommonConfigTemplate {
-    /// Consensus configuration.
-    pub consensus: ConsensusConfig,
-    /// General configuration.
-    pub general: GeneralConfig,
-}
+use std::{
+    net::SocketAddr,
+    path::{Path, PathBuf},
+};
 
 /// Part of the template configuration.
 #[derive(PartialEq, Clone, Debug, Serialize, Deserialize, Default)]
 pub struct GeneralConfig {
     /// Count of the validator nodes in the network.
     pub validators_count: u32,
+    /// Supervisor service mode.
+    pub supervisor_mode: SupervisorMode,
 }
 
-/// Node public configurations.
-#[derive(Debug, Clone, Serialize, Deserialize)]
+/// Public configuration of the node. Is shared among validators.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
 pub struct NodePublicConfig {
-    /// Network address.
-    pub address: String,
-    /// Public keys of a validator.
-    pub validator_keys: ValidatorKeys,
-}
-
-/// `SharedConfig` contain all public information that should be shared in the handshake process.
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct SharedConfig {
-    /// Template for common configuration
-    pub common: CommonConfigTemplate,
-    /// Public node
-    pub node: NodePublicConfig,
+    /// Consensus configuration.
+    pub consensus: ConsensusConfig,
+    /// General configuration.
+    pub general: GeneralConfig,
+    /// Keys of the node.
+    pub validator_keys: Option<ValidatorKeys>,
 }
 
-/// `NodePrivateConfig` collects all public and secret keys.
-#[derive(Debug, Serialize, Deserialize)]
+/// Private configuration of the node.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 pub struct NodePrivateConfig {
-    /// Listen address.
+    /// Network listening address.
     pub listen_address: SocketAddr,
-    /// External address.
+    /// Remote Network address used by this node.
```

Comment:
Not sure about phrasing. What is "Remote Network address" and how is it exactly "used"?

Reply:
🤷‍♂ I just copied docstrings from NodeConfig in core. I have no idea how to describe this field in a proper way. @alekseysidorov ?

Reply:
I also have no ideas for the better naming.